test/test_optim.py

```diff
@@ -512,6 +596,141 @@ def test_reduce_lr_on_plateau8(self):
                                       threshold=0.1, patience=5, cooldown=5)
         self._test_reduce_lr_on_plateau(scheduler, targets, metrics, epochs)
 
+    def test_compound_step_and_multistep_lr(self):
```

Comment:
Out of curiosity, how did you pick which combinations to test? :) (It might also be worth just explicitly testing the old implementations, with no combinations, against the new ones; the new implementations seem more complicated, so it seems helpful to actually test explicitly against the reference implementation, which is simpler.)

Reply:
I added 10 tests. First, I tested all combinations between StepLR, MultiStepLR, ExponentialLR and CosineAnnealing (6 combinations). Then I tested ReduceLROnPlateau + 1 of these modified changes (4 in total). The purpose of these tests is to see that combining different schedulers now works as intended.

For single implementations, there are already uni-tests for old implementations, and the new implementations already pass those tests, so that gives some validation. But, I can add some tests that implements the old codes explicitly for comparison as you suggested.